lib/matplotlib/image.py

```diff
@@ -983,13 +984,14 @@ def get_cursor_data(self, event):
         if self.origin == 'upper':
             ymin, ymax = ymax, ymin
         arr = self.get_array()
-        data_extent = Bbox([[ymin, xmin], [ymax, xmax]])
-        array_extent = Bbox([[0, 0], arr.shape[:2]])
-        trans = BboxTransform(boxin=data_extent, boxout=array_extent)
-        point = trans.transform([event.ydata, event.xdata])
+        data_extent = Bbox([[xmin, ymin], [xmax, ymax]])
```

Comment:
I'm confused - was the previous version just a huge bug?

Reply:
Are you referring to the switched `xdata` and `ydata` coordinates in the original version? I think this was fine originally (this code was covered quite well by tests in `test_cursor_data()`).
The first index of the array runs along the y-axis, while the second index runs along the x-axis. This means that a flip has to happen somewhere.
Originally, the flip was in the line `point = trans.transform([event.ydata, event.xdata])`. However, since we now want to apply a different transform first, we have to input the points in the right order (first x, then y). The flip still has to happen though, so we instead flip `arr.shape` and change `i, j` to `j, i` when we get the result.

Reply:
I got it - I didn't trace it all through so was confused...